### PR TITLE
Add iam_tags to vault_aws_secret_backend_role resource

### PR DESCRIPTION
**Pull request by pporee**
### Description
Fix https://github.com/hashicorp/terraform-provider-vault/issues/1073

Relates OR Closes #0000

### Checklist
- [ ] Added [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md) entry (only for user-facing changes)
- [ ] Acceptance tests where run against all supported Vault Versions

### Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

File: vault/resource_aws_secret_backend_role_test.go [hunk 1]
```diff
@@ -331,6 +371,10 @@ resource "vault_aws_secret_backend_role" "test_role_arns" {
     credential_type = "assumed_role"
     iam_groups = ["group1", "group2"]
     backend = vault_aws_secret_backend.test.path
+	iam_tags = {
+		"key1" = "value1"
+		"key2" = "value2"
+	  }
```
Comment (thyton) [inline vault/resource_aws_secret_backend_role_test.go:377]:
```suggestion
  iam_tags = {
    "key1" = "value1"
    "key2" = "value2"
  }
```

**Review by thyton — CHANGES_REQUESTED** [review]
Hi @pporee, this PR seems to break some of the acceptance tests (see below for an example). Local test setup guides can be found at https://github.com/hashicorp/terraform-provider-vault/blob/main/README.md#developing-the-provider. Could you take a look and drop your acceptance tests' output in the description like [#781's](https://github.com/hashicorp/terraform-provider-vault/pull/781#issue-626071588)?
 
Please let us know if you need any help.

Thanks,
Thy

**Comment by Shocktrooper** [comment]
Hello! I am interested in this because it will allow us to use AWS policy variables in our IAM policies.  Without this we cannot do that and thus cannot use AWS Policy variables based off of tags
- https://docs.aws.amazon.com/IAM/latest/UserGuide/access_tags.html

**Comment by Shocktrooper** [comment]
@thyton Can you check this code again and see if there are any other changes needed other than what you have reviewed since its a few months old now.  if @pporee cannot complete this I can go ahead and try to finish it with a new MR

**Comment by thyton** [comment]
@Shocktrooper Thank you for your patience! I've created a WIP #2231 to finish the remaining work and close this PR.
I'm expecting to close #2231 within 2 weeks.

**Comment by Shocktrooper** [comment]
Thank you for picking this up! @thyton  Let me know if there is anything I can do

**Comment by thyton** [comment]
@pporee @Shocktrooper This feature will be out in release [4.3.0](https://github.com/hashicorp/terraform-provider-vault/milestone/50). Thank you again for your patience!

**Pull request closed without merging**